programs/survival/silk_budding_amethyst.sc

```diff
@@ -0,0 +1,26 @@
+// Conversion of Silk Spwaners; Implemented by Scott Gudeman (DragonCrafted87)
+// See https://github.com/gnembon/scarpet/blob/master/programs/survival/silk_spawners.sc for Original Basis
+
+//keeps script loaded upon server start
+__config() -> {'stay_loaded' -> true, 'scope' -> 'global'};
+
+// private __silk_budding_amethyst (player, block)
+// returns: true if and only if
+//  - player is holding a silk touch diamond or netherite pickaxe
+//  - block is a budding_amethyst
+__silk_budding_amethyst (player, block) -> (
+    [tool, count, nbt] = player ~ 'holds' || ['None', 0, null];
+
+    block=='budding_amethyst' &&
+    tool ~ ['diamond_pickaxe', 'netherite_pickaxe'] &&
+    nbt ~ 'silk_touch';
+);
+
+
+__on_player_breaks_block (player, block) -> (
+    __silk_budding_amethyst(player, block) &&
+    (
+        item_nbt = nbt('{Item:{id:"minecraft:budding_amethyst",Count:1b,PickupDelay:10}}');
```

Comment:
I recommend adding a little velocity, to make it more like what happens when you actually break a block.

Reply:
this was fun to figure out if items didn't despawn naturally I have a feeling I'd be randomly finding the ones that flew off like rockets for a while